### ISSUE TITLE
Segmentation fault error v8.1.23+ when doing partitioned analysis

### ISSUE THREAD

**Issue by davised** [stamatak/standard-RAxML#22]
Hi,

I've worked with v8.1.20 for the past year with good success but recently decided to upgrade to the newest version (v8.2.4) after doing some spring cleaning.  I was able to use the 8.2.4 to do some model testing (with a modified ProteinModelSelection.pl) for 10 individual protein alignments with the intention of doing a partitioned analysis later on.  When I attempted to to the partitioned analysis, I got a Seg fault error with a core dump.  I looked through the commits and saw that you fixed some issues with partitioned analyses in v8.1.23 so I guessed that maybe the error was introduced. I tested my data with v8.1.22 and it can run, but v8.1.23 seg faults.

I'm not sure this is problem specifically with my dataset or more widespread. I can attach sample files if it will be helpful.

Thanks!

Ed Davis

**Comment by willgearty** [comment]
Hi Ed,
I've also been encountering some partition problems with the most recent version.
I was able to resolve the problems by using the sequential version.
Are you using the MPI or Hybrid version of RAxML? If so, could you try running your analysis with the sequential version?
-Will

**Comment by davised** [comment]
Hi Will,

Thanks for commenting as I left out the compiled version that I was using!  My tests were with the SSE3-PTHREADS version. I just tested on the SSE3 version (v8.2.4) and it worked!  There does seem to be a problem somewhat specific to the PTHREADS version, which I didn't catch in my earlier testing.

Unfortunately our cluster is not set up for MPI, so I cannot test the MPI or hybrid versions currently.

So to summarize what I have actually tested:

Non-partitioned analysis is successful on these versions:
SSE3-PTHREADS v8.2.4
SSE3-PTHREADS v8.1.20

Partitioned analysis is successful on these versions:
SSE3 v8.2.4
SSE3-PTHREADS v8.1.22
SSE3-PTHREADS v8.1.20

Partitioned analysis is unsuccessful on these versions:
SSE3-PTHREADS v8.2.4
SSE3-PTHREADS v8.1.23

Regards,

Ed

**Comment by willgearty** [comment]
Hi Ed,
Unfortunately, I don't think much development will be happening until March-ish (I'm just a lowly user), but at least you can still analyse your data using the sequential version of RAxML!
Cheers,
Will

**Comment by stamatak** [comment]
Hi Ed,

Please send me the dataset causing the problem, but as Will said, I 
won't be able to do any coding before March.

Alexis

On 04.02.2016 01:58, Will Gearty wrote:
> Hi Ed,
> Unfortunately, I don't think much development will be happening until
> March-ish (I'm just a lowly user), but at least you can still analyse
> your data using the sequential version of RAxML!
> Cheers,
> Will
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stamatak/standard-RAxML/issues/22#issuecomment-179557416>.
>

-- 
Alexandros (Alexis) Stamatakis

Research Group Leader, Heidelberg Institute for Theoretical Studies
Full Professor, Dept. of Informatics, Karlsruhe Institute of Technology
Adjunct Professor, Dept. of Ecology and Evolutionary Biology, University
of Arizona at Tucson

www.exelixis-lab.org

**Comment by davised** [comment]
Hi Alexis,

I think I've found why the problem exists, and I think it can be replicated independently.  I tried modifying the substitution models in the partitioned dataset and found that I cannot combine the LG4X with other substitution models.  Most of the partitions that I've found do best with LG4X, but one of the genes was better modeled by either LG or MTZOA (very unusual, as these are not mitochondria).  In any case, I can mitigate the problem by excluding LG4X from my substitution model tests, or by only using LG4X for each of the partitions.

Thanks for your time and awesome program!

Ed

**Comment by stamatak** [comment]
Thanks Ed :-) RAxML urgently needs a re-design from scratch though ...

Alexis

On 05.02.2016 00:45, Ed Davis wrote:
> Hi Alexis,
>
> I think I've found why the problem exists, and I think it can be
> replicated independently. I tried modifying the substitution models in
> the partitioned dataset and found that I cannot combine the LG4X with
> other substitution models. Most of the partitions that I've found do
> best with LG4X, but one of the genes was better modeled by either LG or
> MTZOA (very unusual, as these are not mitochondria). In any case, I can
> mitigate the problem by excluding LG4X from my substitution model tests,
> or by only using LG4X for each of the partitions.
>
> Thanks for your time and awesome program!
>
> Ed
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stamatak/standard-RAxML/issues/22#issuecomment-180106985>.
>

-- 
Alexandros (Alexis) Stamatakis

Research Group Leader, Heidelberg Institute for Theoretical Studies
Full Professor, Dept. of Informatics, Karlsruhe Institute of Technology
Adjunct Professor, Dept. of Ecology and Evolutionary Biology, University
of Arizona at Tucson

www.exelixis-lab.org

**Comment by stamatak** [comment]
Problems with Pthreads and LG4X should be solved with version 8.2.7 

alexis
